pkg/catalog/utils/version.go

```diff
@@ -63,6 +65,41 @@ func VersionGreaterThan(a, b string) bool {
 	return version.GreaterThan(a, b)
 }
 
+func ValidateChartCompatibility(template *v3.CatalogTemplateVersion, clusterlister v3.ClusterLister, clusterName string) error {
+	if err := ValidateRancherVersion(template); err != nil {
+		return err
+	}
+	if err := ValidateKubeVersion(template, clusterlister, clusterName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidateKubeVersion(template *v3.CatalogTemplateVersion, clusterlister v3.ClusterLister, clusterName string) error {
+	if template.Spec.KubeVersion == "" {
+		return nil
+	}
+	constraint, err := semver.ParseRange(template.Spec.KubeVersion)
+	if err != nil {
+		logrus.Errorf("failed to parse constraint for kubeversion %s: %v", template.Spec.KubeVersion, err)
+		return nil
```

Comment:
q: why don't we return the error here?

Reply:
Because an error in this function insinuates that the cluster version is not compatible with the chart, but if the KubeVersion constraint is not readable I think it's best to just treat it as though it's empty and install the app. Installing an app that isn't compatible won't be catastrophic.